File: node-client/examples/simple.js
const { NexumClient } = require("..")

const client = new NexumClient({
  host: "http://localhost:3000",
  token: "token"
})

client.on("connect", () => {
  console.log("connected")
})

client.on("disconnect", () => {
  console.log("disconnected")
})

client.on("sync", data => console.log("sync:", data))

client.on("receive", (customId, value) =>
  console.log("received:", { customId, value })
)

let i = 0
setInterval(() => {
  client.update("slider", i)
  i = i > 10 ? 0 : i + 0.2
}, 1000)

File: python-client/examples/simple.py
import nexum

client = nexum.Client(host="http://localhost:3000", token="token")

@client.on
def connect():
  print("connected!")

@client.on("disconnect")
def on_disconnect():
  print("disconnected!")

@client.on("receive")
def receive(custom_id, value):
  print(custom_id, value)

  if custom_id == "button":
    client.update("button", not value)

@client.on("sync")
def sync(data):
  print("sync:", data)


client.connect()
